src/provider/controller/ContextHandler.ts

```diff
@@ -39,7 +39,13 @@ export class ContextHandler {
 
         this._channelModel.setContext(channel.id, context);
 
-        return Promise.all(channelMembers.map(identity => this.send(identity, context))).then(() => {});
+        const sourceId = AppWindow.getId(source);
+
+        return Promise.all(channelMembers
```

Comment:
I know it was like this before, but instead of 
```
return Promise.all(...).then(() => {})
```
could we just do 
```
await Promise.all(...)
```

Reply:
Sure thing, I also find it clearer that way. Will implement all feedback in one go if that's ok

Reply:
Yea, no problem. Hardly a blocker, just something I noticed in passing.